src/legacy/core_plugins/kibana/public/dashboard/dashboard_app.js

```diff
@@ -220,6 +221,7 @@ app.directive('dashboardApp', function ($injector) {
         $scope.model.query = migrateLegacyQuery(query);
         dashboardStateManager.applyFilters($scope.model.query, filterBar.getFilters());
         $scope.refresh();
+        if ($scope.reload) $scope.reload();
```

Comment:
What does this line actually fix?  I tried commenting it out and everything seems to still work.

Reply:
This isn't quite related to this PR, but the line below `$scope.$watch('model.query', $scope.updateQueryAndFetch);` I believe is causing all of this to be called twice.  I think it can be removed, I didn't notice anything going wrong when I did and I suspect it's left over from pre query bar in react days.  If we are going to reload everything, might be worth getting rid of it to prevent the extra reload time.

Reply:
Forget that top comment (I'll go ahead and delete it), I must not have been using a relative date range or something, but I do see this is necessary.